File: code/native/browser/runtime/json.ts
// JSON over the host JSON. The parsed value is the opaque dynamic value (any); the accessors navigate it directly.
const json = {
  parse: (text: string): any => JSON.parse(text),
  stringify: (value: any): string => JSON.stringify(value),
  getField: (value: any, key: string): any => (value != null && typeof value === 'object' && !Array.isArray(value) && key in value) ? value[key] : null,
  getItem: (value: any, index: number): any => Array.isArray(value) && index >= 0 && index < value.length ? value[index] : null,
  asNumber: (value: any): number => typeof value === 'number' ? value : 0,
  asText: (value: any): string => typeof value === 'string' ? value : '',
  asBoolean: (value: any): boolean => typeof value === 'boolean' ? value : false,
  isNull: (value: any): boolean => value === null || value === undefined,
}

File: code/native/rust/runtime/json.rs
mod json {
    use serde_json::Value;
    pub fn parse(text: String) -> Value { serde_json::from_str(&text).unwrap_or(Value::Null) }
    pub fn stringify(value: Value) -> String { serde_json::to_string(&value).unwrap_or_default() }
    pub fn get_field(value: Value, key: String) -> Value { value.get(&key).cloned().unwrap_or(Value::Null) }
    pub fn get_item(value: Value, index: i64) -> Value { value.get(index as usize).cloned().unwrap_or(Value::Null) }
    pub fn as_number(value: Value) -> f64 { value.as_f64().unwrap_or(0.0) }
    pub fn as_text(value: Value) -> String { value.as_str().map(|s| s.to_string()).unwrap_or_default() }
    pub fn as_boolean(value: Value) -> bool { value.as_bool().unwrap_or(false) }
    pub fn is_null(value: Value) -> bool { value.is_null() }
}
